axelor-account/src/main/java/com/axelor/apps/account/service/PartnerTurnoverServiceImpl.java

```diff
@@ -0,0 +1,354 @@
+package com.axelor.apps.account.service;
+
+import com.axelor.apps.account.db.repo.InvoiceRepository;
+import com.axelor.apps.base.db.Currency;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.db.PartnerTurnover;
+import com.axelor.apps.base.db.Year;
+import com.axelor.apps.base.db.repo.CompanyRepository;
+import com.axelor.apps.base.db.repo.PartnerRepository;
+import com.axelor.apps.base.db.repo.PartnerTurnoverRepository;
+import com.axelor.apps.base.service.CurrencyService;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.db.JPA;
+import com.axelor.exception.AxelorException;
+import com.axelor.inject.Beans;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PartnerTurnoverServiceImpl {
+
+  private final Logger log = LoggerFactory.getLogger(PartnerTurnoverServiceImpl.class);
+  @Inject AppBaseService appBaseService;
+  @Inject CurrencyService currencyService;
```

Comment:
Please use injection by constructor to avoid issues with injection in the future.
Also please create an interface PartnerTurnoverService to be implemented by PartnerTurnoverServiceImpl and bind it in AccountModule.

Reply:
Done